src/File/ParentDirectoryRelativePathHelper.php

```diff
@@ -54,6 +54,11 @@ public function getFilenameParts(string $filename): array
 		}
 
 		$dotsCount = $parentPartsCount - $i;
+
+		if ($dotsCount < 0) {
```

Comment:
I had to re-add these lines, which were deleted with https://github.com/phpstan/phpstan-src/pull/637/files#r696555714

it seems it was a false-positive that phpstan could detect this case with the new integer-range math